src/utils.cpp

```diff
@@ -1947,17 +1947,26 @@ void magic_loss(struct char_data *ch, int magic, bool msg)
 }
 
 // Return true if the character has a kit of the given type, false otherwise.
-#define IS_KIT(obj, type) (GET_OBJ_TYPE((obj)) == ITEM_WORKSHOP && GET_WORKSHOP_TYPE((obj)) == type && GET_WORKSHOP_GRADE((obj)) == TYPE_KIT)
+// Note: some kits may still have workshop grade of 0 instead of TYPE_KIT.
+#define IS_KIT(obj, type) ( GET_OBJ_TYPE((obj)) == ITEM_WORKSHOP && GET_WORKSHOP_TYPE((obj)) == type && (GET_WORKSHOP_GRADE((obj)) == TYPE_KIT || GET_WORKSHOP_GRADE((obj)) == 0) )
 bool has_kit(struct char_data * ch, int type)
 {
-  for (struct obj_data *o = ch->carrying; o; o = o->next_content)
-    if (IS_KIT(o, type))
-      return TRUE;
+  struct obj_data *obj;
+  bool found = FALSE;
+  
+  for (obj = ch->carrying; obj && !found; obj = obj->next_content) {
```

Comment:
Since `obj` isn't being used outside of this scope, let's go ahead and keep the `struct obj_data *obj` declaration line in here as well. Declaring a variable outside of the scope it's used leads to bugs like the one you saw with `int i` above.

Reply:
Can't test it right now, but a question: when simultaneously declaring and initializing obj in the for loop, should it be:
> struct obj_data *obj = ch->carrying
or
> struct obj_data *obj = &(ch->carrying)

Reply:
I can never seem to get pointers and references right >.<

Going with:
> struct obj_data *obj = ch->carrying

Since that's how it's done everywhere else in the code.